packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -11410,6 +11436,205 @@ describe('NetworkController', () => {
     });
   });
 
+  describe('dangerouslySetNetworkConfiguration', () => {
```

Comment:
This test file is large. Understandably since the network controller is responsible for MANY things and logic.

I've tried to keep these tests isolated and lean.